### PR TITLE
Add Button chaser (Normal) 

### PR DESCRIPTION
**Pull request by MarcRaeder**
Add Button-Chaser without Bonus-tasks

### CODE REVIEW

File: button-chaser.css [hunk 1]
```diff
@@ -0,0 +1,49 @@
+.project-container {
+    width: auto;
+    height: 860px;
+}
```
Comment (stefanluth) [inline button-chaser.css]: I believe .css files should be indented with 2 whitespaces, not 4.

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "html.format.indentInnerHtml": true
+}
```
Comment (stefanluth) [inline .vscode/settings.json]: Remove :)

File: button-chaser.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+const Movement = 40;
+button = document.getElementsByClassName("button-container")[0];
```
Comment (stefanluth) [inline button-chaser.js]:
```suggestion
const button = document.getElementsByClassName("button-container")[0];
```

File: button-chaser.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        if (mouseIsOnRightTop) {
+            return mousePosition.rightTop;
+        }
+        if (mouseIsOnLeftBottom) {
+            return mousePosition.leftBottom;
+        }
+        if (mouseIsOnRightBottom) {
+            return mousePosition.rightBottom;
+        }
+    }
```
Comment (stefanluth) [inline button-chaser.js]: You could move function outside the moveButton function.

**Review by stefanluth — CHANGES_REQUESTED** [review]

**Comment by stefanluth** [comment]
I believe these (literal) edge cases should behave differently, no?

https://user-images.githubusercontent.com/80166473/188649086-4f9fc17c-8a57-476f-8907-f97bfd45ad00.mov


https://user-images.githubusercontent.com/80166473/188649308-7ff37e88-7f5e-407b-b5c6-a95518e75f79.mov

I would expect the button to go away from the edge, if my cursor is coming from there.

File: button-chaser.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <div class="project-container">
+      <div class=" button-container">
+        <button class="button-to-chase">Klick mich!</button>
+      </div>
+    </div>
+  </body>
+
+  <script src="button-chaser.js"></script>
+
+</html>
```
Comment (stefanluth) [inline button-chaser.html:22]: Two spaces indents :)

File: button-chaser.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+};
+
+function getMousePosition(event) {
+
+    const buttonTopSide = buttonContainer.offsetTop + button.offsetTop;
+    const buttonLeftSide = buttonContainer.offsetLeft + button.offsetLeft;
+    const buttonRightSide = buttonLeftSide + button.offsetWidth;
+    const buttonBottomSide = buttonTopSide + button.offsetHeight;
+
+    const mousePositionX = event.x;
```
Comment (stefanluth) [inline button-chaser.js]: Two spaces indents :)

**Review by stefanluth — APPROVED** [review]

**Review by MKrips — APPROVED** [review]

**Pull request merged**